src/components/mixins/UtilityMixin.vue

```diff
@@ -39,7 +44,32 @@ export default class UtilityMixin extends Vue {
                 const profile = this.$store.getters['profile/activeProfile'].asImmutableProfile();
                 const combo = throwForR2Error(await ThunderstoreCombo.fromProtocol(protocolUrl, game));
 
-                await this.$store.dispatch('download/downloadAndInstallSpecific', {profile, combo});
+                const downloadId = await this.$store.dispatch(
```

Comment:
UtilityMixin is still a Vue component so I consider this a step backwards in our efforts to remove the business logic from the UI components. Keeping the Vuex between the UI and logic would make sense to me.